### PR TITLE
add jinja2 global to reserved names

### LINKED ISSUE #41955: jinja keyword collision for variable names
##### SUMMARY

k8s_raw fails using jinja default filter to default undefined variables

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
k8s_raw

##### ANSIBLE VERSION

```
ansible 2.6.0rc4
(python 3)
```

##### STEPS TO REPRODUCE
```
---
- name: K8s
  hosts: localhost
  tasks:
  - name: Create namespace
    k8s:
      name: "{{ namespace | default('horizontesting123', true) }}"
      api_version: v1
      kind: Namespace
      state: present
```
This will fail, because the `name:` does not get the default value.

```
TASK [Create namespace] *****************************************************************************************************************************************************************************************
task path: /Users/proto/SORGENTI/SWITCH-openstack-horizon-k8s-deployment/ansible/test.yaml:7
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: proto
<127.0.0.1> EXEC /bin/sh -c 'echo ~proto && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /Users/proto/.ansible/tmp/ansible-tmp-1530002441.681397-275101092917815 `" && echo ansible-tmp-1530002441.681397-275101092917815="` echo /Users/proto/.ansible/tmp/ansible-tmp-1530002441.681397-275101092917815 `" ) && sleep 0'
Using module file /usr/local/lib/python3.6/site-packages/ansible/modules/clustering/k8s/k8s.py
<127.0.0.1> PUT /Users/proto/.ansible/tmp/ansible-local-308812tchkxm/tmpco_jq671 TO /Users/proto/.ansible/tmp/ansible-tmp-1530002441.681397-275101092917815/k8s.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /Users/proto/.ansible/tmp/ansible-tmp-1530002441.681397-275101092917815/ /Users/proto/.ansible/tmp/ansible-tmp-1530002441.681397-275101092917815/k8s.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/local/opt/python/bin/python3.6 /Users/proto/.ansible/tmp/ansible-tmp-1530002441.681397-275101092917815/k8s.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /Users/proto/.ansible/tmp/ansible-tmp-1530002441.681397-275101092917815/ > /dev/null 2>&1 && sleep 0'
The full traceback is:
  File "/var/folders/10/99r84g5d0_n5ct1248_dmtgh00012p/T/ansible__6rt22sd/ansible_modlib.zip/ansible/module_utils/k8s/raw.py", line 111, in perform_action
    existing = resource.get(name=name, namespace=namespace)
  File "/usr/local/Cellar/python/3.6.5/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/openshift/dynamic/client.py", line 33, in inner
    raise api_exception(e)

fatal: [localhost]: FAILED! => {
    "changed": false,
    "error": 400,
    "invocation": {
        "module_args": {
            "api_key": null,
            "cert_file": null,
            "context": null,
            "host": null,
            "key_file": null,
            "kubeconfig": null,
            "password": null,
            "ssl_ca_cert": null,
            "username": null,
            "verify_ssl": null
        }
    },
    "msg": "Failed to retrieve requested object: b'400 Bad Request'",
    "reason": "Bad Request",
    "status": 400
}
	to retry, use: --limit @/Users/proto/SORGENTI/SWITCH-openstack-horizon-k8s-deployment/ansible/test.retry
```

I have found that doing:
```
---
- name: K8s
  hosts: localhost
  tasks:
  - name: Create namespace
    k8s:
      name: "horizontesting123"
      api_version: v1
      kind: Namespace
      state: present
```
or 
```
---
- name: K8s
  hosts: localhost
  vars:
     namespace: mytest234
  tasks:
  - name: Create namespace
    k8s:
      name: "{{ namespace }}"
      api_version: v1
      kind: Namespace
      state: present
```
then everything works as expected.

##### EXPECTED RESULTS

It looks like the filter `"{{ namespace | default('horizontesting123', true) }}"` does not work in the `name`. Not sure if this is an ansible bug or a k8s_raw module bug. Is there anything wrong in using the default filter there of an undefined variable ?

I would expect the everything to work when using the jinja filter to default undefined variables.

### PR DESCRIPTION
**Pull request by bcoca**
also allow expansion by additional context provided from caller

  fixes #41955

##### ISSUE TYPE

- Bugfix Pull Request
##### COMPONENT NAME

vars

### CODE REVIEW

File: changelogs/fragments/moar_reserved_vars.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - Handle more varnames that can create conflicts, expand function in general, handle jinja2 globasl in particular.
```
Comment (sivel) [inline changelogs/fragments/moar_reserved_vars.yml]:
```suggestion
  - Handle more varnames that can create conflicts, expand function in general, handle jinja2 globals in particular.
```

Comment (Akasurde) [inline changelogs/fragments/moar_reserved_vars.yml]:
```suggestion
  - Handle more varnames that can create conflicts, expand a function in general, handle jinja2 globals in particular (https://github.com/ansible/ansible/issues/41955).
```

**Comment by Akasurde** [comment]
https://github.com/ansible/ansible/pull/57812

**Comment by Akasurde** [comment]
Similar issue - https://github.com/ansible/ansible/issues/57808

**Review by mkrizek — APPROVED** [review]

**Pull request merged**